lib/screens/events/events.dart

```diff
@@ -45,25 +45,23 @@ class _EventsScreenState extends State<EventsScreen> {
     );
   }
 
-  Widget _buildSuccess(context) {
-    Events event = Events.fromJson(json[0]);
+  Widget _buildSuccess(BuildContext context, List<Event> json) {
     //TODO On success UI
     return Center(
       child: Column(
         mainAxisSize: MainAxisSize.min,
-        children: <Widget>[
-          Icon(Icons.check_circle_outline),
-          Text(
-            event.details,
-            textAlign: TextAlign.center,
-          ),
-        ],
+        children: <Widget>[Icon(Icons.check_circle_outline), Text(json[0].name)],
       ),
     );
   }
 
-  void _events() {
+  Widget _buildInitial() {
+    _getAllEvents();
```

Comment:
you are already calling this once (in the listener). You don't have to call this in here.
```suggestion
```